File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {        
//         놀이기구의 원래 이용료가 price원이다.
//         n번째 이용 시 n배 받음.
//         count번 탔을 때 모자란 금액은?
//         부족하지 않으면 0을 리턴
        long answer = money;
        for(var i = 1; i <= count; i++){
            answer -= price * i;
        }
        
        return (answer > 0)?0:-answer;
    }
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.kt
class Solution {
    fun solution(price: Int, money: Int, count: Int): Long {
//         놀이기구의 원래 이용료가 price원이다.
//         n번째 이용 시 n배 받음.
//         count번 탔을 때 모자란 금액은?
//         부족하지 않으면 0을 리턴
        var answer = money.toLong()
        for(i in 1..count){
            answer -= price * i
        }
        return if (answer > 0) 0 else -answer
    }
}